### PR TITLE
Move the CUDA implementation of ceil to ATen.

### LINKED ISSUE #24542: Migrate `ceil` and `ceil_` from the TH to Aten (CUDA)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

### PR DESCRIPTION
**Pull request by xuhdev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25372 Move the CUDA implementation of floor to ATen.
* #25285 Move the CUDA implementation of rsqrt to ATen.
* #25041 Move the CUDA implementation of round to ATen.
* #24879 Create helpers for implementing unary ops whose CUDA implementation is ATen.
* **#24866 Move the CUDA implementation of ceil to ATen.**

Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

### CODE REVIEW

**Comment by xuhdev** [comment]
@VitalyFedyunin This PR should be ready for review now

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 1]
```diff
@@ -32,6 +32,14 @@ void logical_not_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void ceil_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "ceil_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return std::ceil(a);
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: `std::ceil` operates with `double` we need to use `ceilf` for floats for performance reasons.

**Review by VitalyFedyunin — COMMENTED** [review]
Please check usage `ceil` vs `ceilf`

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu (hunk 1 above)
Comment (xuhdev) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: I thought `std::ceil` is overloaded for different floating point types? See https://en.cppreference.com/w/cpp/numeric/math/ceil

Comment (VitalyFedyunin) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: Ah, right, overloaded in C++11. Good point.

**Review by VitalyFedyunin — APPROVED** [review]

**Pull request closed without merging**